mantidproject/mantid
# Adding ads observer to muon analysis

**Description of work.**

This branch adds an ADS observer to the Muon Analysis GUI, so that workspaces deleted in the ADS are deleted gracefully from the GUI

**To test:**

  * Open Muon Analysis 2 load some data, do a fit and create a results table.
  * Delete one of your raw data workspaces, check that the selected runs are amended to no longer include this run and that on the phase tab this workspace no longer appears as an option.
  * Delete your currently selected group/pair, check that this is removed from the plot and from the fitting options. Try a new fit and check this still works
  * Delete one of your fitting workspaces, check that the fit it corresponds to is removed from the results tab.

  * In frequency domain analysis repeat the same steps and check that the options in the transform tab are updated correctly.

 * Check that renaming of work-spaces is treated the same as deletion and that  clearing the ADS clears all data from the GUI.



Fixes #25931 

*This does not require release notes* because this is a new interface which is being added to the release notes elsewhere

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
